### ISSUE TITLE
[BUG]: error: ‘TimingType’ was not declared in this scope

### ISSUE THREAD

**Issue by carlosal1015** [FEniCS/dolfinx#2645]
### How to reproduce the bug

Hi, I guess the problem arises in the newer GCC 13.

```
[  9%] Building CXX object dolfinx/CMakeFiles/dolfinx.dir/common/Table.cpp.o
In file included from /tmp/makepkg/dolfinx/src/dolfinx-0.6.0/cpp/dolfinx/common/MPI.h:16,
                 from /tmp/makepkg/dolfinx/src/dolfinx-0.6.0/cpp/dolfinx/common/MPI.cpp:7:
/tmp/makepkg/dolfinx/src/dolfinx-0.6.0/cpp/dolfinx/graph/AdjacencyList.h:172:14: warning: testing if a concept-id is a valid expression; add ‘requires’ to check satisfaction [-Wmissing-requires]
  172 |              std::convertible_to<
      |              ^
      |              requires 
In file included from /tmp/makepkg/dolfinx/src/dolfinx-0.6.0/cpp/dolfinx/common/MPI.h:16,
                 from /tmp/makepkg/dolfinx/src/dolfinx-0.6.0/cpp/dolfinx/common/IndexMap.h:11,
                 from /tmp/makepkg/dolfinx/src/dolfinx-0.6.0/cpp/dolfinx/common/IndexMap.cpp:7:
/tmp/makepkg/dolfinx/src/dolfinx-0.6.0/cpp/dolfinx/graph/AdjacencyList.h:172:14: warning: testing if a concept-id is a valid expression; add ‘requires’ to check satisfaction [-Wmissing-requires]
  172 |              std::convertible_to<
      |              ^
      |              requires 
In file included from /tmp/makepkg/dolfinx/src/dolfinx-0.6.0/cpp/dolfinx/common/MPI.h:16,
                 from /tmp/makepkg/dolfinx/src/dolfinx-0.6.0/cpp/dolfinx/common/Table.cpp:9:
/tmp/makepkg/dolfinx/src/dolfinx-0.6.0/cpp/dolfinx/graph/AdjacencyList.h:172:14: warning: testing if a concept-id is a valid expression; add ‘requires’ to check satisfaction [-Wmissing-requires]
  172 |              std::convertible_to<
      |              ^
      |              requires 
[ 11%] Building CXX object dolfinx/CMakeFiles/dolfinx.dir/common/Timer.cpp.o
In file included from /tmp/makepkg/dolfinx/src/dolfinx-0.6.0/cpp/dolfinx/common/TimeLogger.h:10,
                 from /tmp/makepkg/dolfinx/src/dolfinx-0.6.0/cpp/dolfinx/common/Timer.cpp:9:
/tmp/makepkg/dolfinx/src/dolfinx-0.6.0/cpp/dolfinx/common/timing.h:22:6: warning: elaborated-type-specifier for a scoped enum must not use the ‘class’ keyword
   22 | enum class TimingType : std::int32_t
      | ~~~~ ^~~~~
      |      -----
/tmp/makepkg/dolfinx/src/dolfinx-0.6.0/cpp/dolfinx/common/timing.h:22:12: error: use of enum ‘TimingType’ without previous declaration
   22 | enum class TimingType : std::int32_t
      |            ^~~~~~~~~~
/tmp/makepkg/dolfinx/src/dolfinx-0.6.0/cpp/dolfinx/common/timing.h:22:23: error: expected unqualified-id before ‘:’ token
   22 | enum class TimingType : std::int32_t
      |                       ^
/tmp/makepkg/dolfinx/src/dolfinx-0.6.0/cpp/dolfinx/common/timing.h:33:24: error: ‘TimingType’ was not declared in this scope
   33 | Table timings(std::set<TimingType> type);
      |                        ^~~~~~~~~~
/tmp/makepkg/dolfinx/src/dolfinx-0.6.0/cpp/dolfinx/common/timing.h:33:34: error: template argument 1 is invalid
   33 | Table timings(std::set<TimingType> type);
      |                                  ^
/tmp/makepkg/dolfinx/src/dolfinx-0.6.0/cpp/dolfinx/common/timing.h:33:34: error: template argument 2 is invalid
/tmp/makepkg/dolfinx/src/dolfinx-0.6.0/cpp/dolfinx/common/timing.h:33:34: error: template argument 3 is invalid
/tmp/makepkg/dolfinx/src/dolfinx-0.6.0/cpp/dolfinx/common/timing.h:41:43: error: ‘TimingType’ was not declared in this scope
   41 | void list_timings(MPI_Comm comm, std::set<TimingType> type,
      |                                           ^~~~~~~~~~
/tmp/makepkg/dolfinx/src/dolfinx-0.6.0/cpp/dolfinx/common/timing.h:41:53: error: template argument 1 is invalid
   41 | void list_timings(MPI_Comm comm, std::set<TimingType> type,
      |                                                     ^
/tmp/makepkg/dolfinx/src/dolfinx-0.6.0/cpp/dolfinx/common/timing.h:41:53: error: template argument 2 is invalid
/tmp/makepkg/dolfinx/src/dolfinx-0.6.0/cpp/dolfinx/common/timing.h:41:53: error: template argument 3 is invalid
/tmp/makepkg/dolfinx/src/dolfinx-0.6.0/cpp/dolfinx/common/TimeLogger.h:42:26: error: ‘TimingType’ was not declared in this scope
   42 |   Table timings(std::set<TimingType> type);
      |                          ^~~~~~~~~~
/tmp/makepkg/dolfinx/src/dolfinx-0.6.0/cpp/dolfinx/common/TimeLogger.h:42:36: error: template argument 1 is invalid
   42 |   Table timings(std::set<TimingType> type);
      |                                    ^
/tmp/makepkg/dolfinx/src/dolfinx-0.6.0/cpp/dolfinx/common/TimeLogger.h:42:36: error: template argument 2 is invalid
/tmp/makepkg/dolfinx/src/dolfinx-0.6.0/cpp/dolfinx/common/TimeLogger.h:42:36: error: template argument 3 is invalid
/tmp/makepkg/dolfinx/src/dolfinx-0.6.0/cpp/dolfinx/common/TimeLogger.h:42:17: error: ‘std::type’ has not been declared
   42 |   Table timings(std::set<TimingType> type);
      |                 ^~~
/tmp/makepkg/dolfinx/src/dolfinx-0.6.0/cpp/dolfinx/common/TimeLogger.h:49:45: error: ‘TimingType’ was not declared in this scope
   49 |   void list_timings(MPI_Comm comm, std::set<TimingType> type,
      |                                             ^~~~~~~~~~
/tmp/makepkg/dolfinx/src/dolfinx-0.6.0/cpp/dolfinx/common/TimeLogger.h:49:55: error: template argument 1 is invalid
   49 |   void list_timings(MPI_Comm comm, std::set<TimingType> type,
      |                                                       ^
/tmp/makepkg/dolfinx/src/dolfinx-0.6.0/cpp/dolfinx/common/TimeLogger.h:49:55: error: template argument 2 is invalid
/tmp/makepkg/dolfinx/src/dolfinx-0.6.0/cpp/dolfinx/common/TimeLogger.h:49:55: error: template argument 3 is invalid
/tmp/makepkg/dolfinx/src/dolfinx-0.6.0/cpp/dolfinx/common/TimeLogger.h:49:36: error: ‘std::type’ has not been declared
   49 |   void list_timings(MPI_Comm comm, std::set<TimingType> type,
      |                                    ^~~
make[2]: *** [dolfinx/CMakeFiles/dolfinx.dir/build.make:146: dolfinx/CMakeFiles/dolfinx.dir/common/Timer.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/Makefile2:304: dolfinx/CMakeFiles/dolfinx.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
```

### Version

0.6.0

### Installation

From the source [(script)](https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=dolfinx#n19).

### Additional information

OS: Arch Linux
Toolchain: [GNU Compiler Collection 13.1.1](https://archlinux.org/packages/core/x86_64/gcc)
[boost](https://archlinux.org/packages/extra/x86_64/boost) version 1.81.0
[petsc](https://aur.archlinux.org/packages/petsc) version 3.19.1
[ffcx](https://aur.archlinux.org/packages/python-fenics-ffcx) version 0.6.0
[cmake](https://archlinux.org/packages/extra/x86_64/cmake) version 3.26.3

**Comment by garth-wells** [comment]
I think the first error was fixed in the dev version by #2513, and I've created a new PR (#2646) that hopefully fixes the second error (on `TimingType`).

Could you test the dev version once #2646 is merged?

**Comment by carlosal1015** [comment]
Thanks, the first error is [fixed](https://aur.archlinux.org/cgit/aur.git/commit/?h=dolfinx&id=b6323f514c51846af10b9aae5b7c3341698e3301).
```
==> Retrieving sources...
  -> Downloading dolfinx-0.6.0.tar.gz...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100  668k    0  668k    0     0  1698k      0 --:--:-- --:--:-- --:--:-- 5553k
  -> Downloading requires-fix.patch...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100  5658    0  5658    0     0  19330      0 --:--:-- --:--:-- --:--:-- 19330
  -> Downloading enum-fix.patch...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100   739    0   739    0     0   3126      0 --:--:-- --:--:-- --:--:--  3126
==> Validating source files with sha512sums...
    dolfinx-0.6.0.tar.gz ... Passed
    requires-fix.patch ... Passed
    enum-fix.patch ... Passed
==> Removing existing $srcdir/ directory...
==> Extracting sources...
  -> Extracting dolfinx-0.6.0.tar.gz with bsdtar
==> Starting prepare()...
patching file cpp/dolfinx/fem/utils.h
patching file cpp/dolfinx/graph/AdjacencyList.h
patching file cpp/dolfinx/fem/utils.h
patching file cpp/dolfinx/common/timing.h
==> Starting build()...
-- The C compiler identification is GNU 13.1.1
-- The CXX compiler identification is GNU 13.1.1
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/sbin/gcc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/sbin/g++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found MPI_C: /usr/lib/libmpi.so (found suitable version "3.1", minimum required is "3") 
-- Found MPI_CXX: /usr/lib/libmpi_cxx.so (found suitable version "3.1", minimum required is "3") 
-- Found MPI: TRUE (found suitable version "3.1", minimum required is "3")  
-- Performing Test HAVE_PIPE
-- Performing Test HAVE_PIPE - Success
-- Performing Test HAVE_PEDANTIC
-- Performing Test HAVE_PEDANTIC - Success
-- Performing Test HAVE_DEBUG
-- Performing Test HAVE_DEBUG - Success
-- Performing Test HAVE_O2_OPTIMISATION
-- Performing Test HAVE_O2_OPTIMISATION - Success
-- Found Boost 1.81.0 at /usr/lib/cmake/Boost-1.81.0
--   Requested configuration: QUIET REQUIRED COMPONENTS timer
-- Found boost_headers 1.81.0 at /usr/lib/cmake/boost_headers-1.81.0
-- Found boost_timer 1.81.0 at /usr/lib/cmake/boost_timer-1.81.0
--   [x] libboost_timer.so.1.81.0
--   [ ] libboost_timer.a
-- Adding boost_timer dependencies: chrono;headers
-- Found boost_chrono 1.81.0 at /usr/lib/cmake/boost_chrono-1.81.0
--   [x] libboost_chrono.so.1.81.0
--   [ ] libboost_chrono.a
-- Adding boost_chrono dependencies: headers
-- Found Boost: /usr/lib/cmake/Boost-1.81.0/BoostConfig.cmake (found suitable version "1.81.0", minimum required is "1.70") found components: timer 
-- Found Python3: /usr/sbin/python3.11 (found version "3.11.3") found components: Interpreter 
-- Adding /usr/lib/python3.11/site-packages/basix to Basix search hints
-- Found PkgConfig: /usr/sbin/pkg-config (found version "1.8.1") 
-- Checking for one of the modules 'PETSc>=3.15;petsc>=3.15'
-- Looking for PETSC_USE_COMPLEX
-- Looking for PETSC_USE_COMPLEX - not found
-- Found PETSc version 3.19.1, prefix: /opt/petsc/linux-c-opt
-- HDF5: Using hdf5 compiler wrapper to determine C configuration
-- Found HDF5: /usr/lib/libhdf5.so;/usr/lib/libsz.so;/usr/lib/libz.so;/usr/lib/libdl.a;/usr/lib/libm.so (found version "1.14.0") found components: C 
-- HDF5_DIR: HDF5_DIR-NOTFOUND
-- HDF5_DEFINITIONS: 
-- HDF5_INCLUDE_DIRS: /usr/include
-- HDF5_LIBRARIES: /usr/lib/libhdf5.so;/usr/lib/libsz.so;/usr/lib/libz.so;/usr/lib/libdl.a;/usr/lib/libm.so
-- HDF5_HL_LIBRARIES: 
-- HDF5_C_DEFINITIONS: 
-- HDF5_C_INCLUDE_DIR: 
-- HDF5_C_INCLUDE_DIRS: /usr/include
-- HDF5_C_LIBRARY: 
-- HDF5_C_LIBRARIES: /usr/lib/libhdf5.so;/usr/lib/libsz.so;/usr/lib/libz.so;/usr/lib/libdl.a;/usr/lib/libm.so
-- HDF5_C_HL_LIBRARY: 
-- HDF5_C_HL_LIBRARIES: 
-- Defined targets (if any):
-- ... hdf5::hdf5
-- Asking Python module FFCx for location of UFC... (Python executable: /usr/sbin/python3.11)
-- Found UFCx: /usr/lib/python3.11/site-packages/ffcx/codegeneration (found suitable version "0.6.0", minimum required is "0.6") 
CMake Warning at CMakeLists.txt:273 (find_package):
  By not providing "FindADIOS2.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "ADIOS2", but
  CMake did not find one.
  Could not find a package configuration file provided by "ADIOS2" (requested
  version 2.8.1) with any of the following names:
    ADIOS2Config.cmake
    adios2-config.cmake
  Add the installation prefix of "ADIOS2" to CMAKE_PREFIX_PATH or set
  "ADIOS2_DIR" to a directory containing one of the above files.  If "ADIOS2"
  provides a separate development package or SDK, be sure it has been
  installed.
-- Checking for one of the modules 'slepc>=3.15'
-- Checking for package 'SCOTCH-PT'
-- SCOTCH could not be found. Be sure to set SCOTCH_ROOT. (missing: SCOTCH_LIBRARIES SCOTCH_INCLUDE_DIRS SCOTCH_TEST_RUNS) 
-- Performing Test PARMETIS_TEST_RUNS
-- Performing Test PARMETIS_TEST_RUNS - Success
-- Found ParMETIS: /lib/libparmetis.so;/lib/libmetis.so (Required is at least version "4.0.2") 
-- Checking for package 'KaHIP'
-- KaHIP could not be found/configured. (missing: KAHIP_INCLUDE_DIRS KAHIP_LIBRARIES KAHIP_TEST_RUNS) 
-- The following features have been enabled:
 * BUILD_SHARED_LIBS, Build DOLFINx with shared libraries.
 * CMAKE_INSTALL_RPATH_USE_LINK_PATH, Add paths to linker search and installed rpath.
 * DOLFINX_UFCX_PYTHON, Enable UFCx discovery using Python. Disable if UFCx should be found using a CMake config file.
-- The following RECOMMENDED packages have been found:
 * ParMETIS (required version >= 4.0.2), Parallel Graph Partitioning and Fill-reducing Matrix Ordering, <http://glaros.dtc.umn.edu/gkhome/metis/parmetis/overview>
   Parallel graph partitioning
-- The following REQUIRED packages have been found:
 * MPI (required version >= 3)
 * pugixml
 * boost_chrono (required version == 1.81.0)
 * boost_headers (required version == 1.81.0)
 * boost_timer (required version == 1.81.0)
 * Boost (required version >= 1.70), Boost C++ libraries, <http://www.boost.org>
 * Basix (required version >= 0.6)
 * PETSc, Portable, Extensible Toolkit for Scientific Computation (PETSc), <https://www.mcs.anl.gov/petsc/>
   PETSc linear algebra backend
 * HDF5, Hierarchical Data Format 5 (HDF5), <https://www.hdfgroup.org/HDF5>
 * Python3
 * UFCx (required version >= 0.6), Interface for form-compilers (part of FFCx), <https://github.com/fenics/ffcx>
 * PkgConfig
-- The following features have been disabled:
 * DOLFINX_SKIP_BUILD_TESTS, Skip build tests for testing usability of dependency packages.
-- The following OPTIONAL packages have not been found:
 * ADIOS2 (required version >= 2.8.1), Adaptable Input/Output (I/O) System., <https://adios2.readthedocs.io/en/latest/>
   IO, including in parallel
 * SCOTCH, Programs and libraries for graph, mesh and hypergraph partitioning, <https://www.labri.fr/perso/pelegrin/scotch>
   Parallel graph partitioning
 * KaHIP, A family of graph partitioning programs, <https://kahip.github.io/>
   Parallel graph partitioning
-- The following RECOMMENDED packages have not been found:
 * SLEPc, Scalable Library for Eigenvalue Problem Computations, <http://slepc.upv.es/>
   Eigenvalue computation
-- 
-- Copying demo and test data to build directory.
-- ----------------------------------------------
-- Configuring done (7.2s)
-- Generating done (0.0s)
-- Build files have been written to: /tmp/makepkg/dolfinx/src/build
[  7%] Building CXX object dolfinx/CMakeFiles/dolfinx.dir/common/defines.cpp.o
[  7%] Building CXX object dolfinx/CMakeFiles/dolfinx.dir/common/IndexMap.cpp.o
[  7%] Building CXX object dolfinx/CMakeFiles/dolfinx.dir/common/log.cpp.o
[  7%] Building CXX object dolfinx/CMakeFiles/dolfinx.dir/common/MPI.cpp.o
[  9%] Building CXX object dolfinx/CMakeFiles/dolfinx.dir/common/Table.cpp.o
[ 11%] Building CXX object dolfinx/CMakeFiles/dolfinx.dir/common/Timer.cpp.o
[ 13%] Building CXX object dolfinx/CMakeFiles/dolfinx.dir/common/TimeLogger.cpp.o
[ 15%] Building CXX object dolfinx/CMakeFiles/dolfinx.dir/common/TimeLogManager.cpp.o
[ 17%] Building CXX object dolfinx/CMakeFiles/dolfinx.dir/common/timing.cpp.o
[ 19%] Building CXX object dolfinx/CMakeFiles/dolfinx.dir/fem/DirichletBC.cpp.o
[ 21%] Building CXX object dolfinx/CMakeFiles/dolfinx.dir/fem/CoordinateElement.cpp.o
[ 23%] Building CXX object dolfinx/CMakeFiles/dolfinx.dir/fem/DofMap.cpp.o
[ 25%] Building CXX object dolfinx/CMakeFiles/dolfinx.dir/fem/ElementDofLayout.cpp.o
[ 26%] Building CXX object dolfinx/CMakeFiles/dolfinx.dir/fem/FiniteElement.cpp.o
[ 28%] Building CXX object dolfinx/CMakeFiles/dolfinx.dir/fem/FunctionSpace.cpp.o
[ 30%] Building CXX object dolfinx/CMakeFiles/dolfinx.dir/fem/dofmapbuilder.cpp.o
[ 32%] Building CXX object dolfinx/CMakeFiles/dolfinx.dir/fem/interpolate.cpp.o
[ 34%] Building CXX object dolfinx/CMakeFiles/dolfinx.dir/fem/petsc.cpp.o
[ 36%] Building CXX object dolfinx/CMakeFiles/dolfinx.dir/fem/sparsitybuild.cpp.o
In file included from /usr/include/c++/13.1.1/x86_64-pc-linux-gnu/bits/c++allocator.h:33,
                 from /usr/include/c++/13.1.1/bits/allocator.h:46,
                 from /usr/include/c++/13.1.1/bits/alloc_traits.h:39,
                 from /usr/include/c++/13.1.1/ext/alloc_traits.h:34,
                 from /usr/include/c++/13.1.1/bits/hashtable_policy.h:39,
                 from /usr/include/c++/13.1.1/bits/hashtable.h:35,
                 from /usr/include/c++/13.1.1/bits/unordered_map.h:33,
                 from /usr/include/c++/13.1.1/unordered_map:41,
                 from /usr/include/c++/13.1.1/functional:63,
                 from /tmp/makepkg/dolfinx/src/dolfinx-0.6.0/cpp/dolfinx/fem/dofmapbuilder.h:10,
                 from /tmp/makepkg/dolfinx/src/dolfinx-0.6.0/cpp/dolfinx/fem/dofmapbuilder.cpp:7:
In member function ‘void std::__new_allocator<_Tp>::deallocate(_Tp*, size_type) [with _Tp = int]’,
    inlined from ‘constexpr void std::allocator< <template-parameter-1-1> >::deallocate(_Tp*, std::size_t) [with _Tp = int]’ at /usr/include/c++/13.1.1/bits/allocator.h:210:35,
    inlined from ‘static constexpr void std::allocator_traits<std::allocator<_Up> >::deallocate(allocator_type&, pointer, size_type) [with _Tp = int]’ at /usr/include/c++/13.1.1/bits/alloc_traits.h:516:23,
    inlined from ‘constexpr void std::_Vector_base<_Tp, _Alloc>::_M_deallocate(pointer, std::size_t) [with _Tp = int; _Alloc = std::allocator<int>]’ at /usr/include/c++/13.1.1/bits/stl_vector.h:387:19,
    inlined from ‘constexpr void std::vector<_Tp, _Alloc>::reserve(size_type) [with _Tp = int; _Alloc = std::allocator<int>]’ at /usr/include/c++/13.1.1/bits/vector.tcc:93:17,
    inlined from ‘std::pair<std::vector<long int>, std::vector<int> > {anonymous}::get_global_indices(const dolfinx::mesh::Topology&, int32_t, int64_t, const std::vector<long int>&, const std::vector<int>&, const std::vector<std::pair<signed char, int> >&)’ at /tmp/makepkg/dolfinx/src/dolfinx-0.6.0/cpp/dolfinx/fem/dofmapbuilder.cpp:481:24,
    inlined from ‘std::tuple<dolfinx::common::IndexMap, int, dolfinx::graph::AdjacencyList<int> > dolfinx::fem::build_dofmap_data(MPI_Comm, const dolfinx::mesh::Topology&, const ElementDofLayout&, const std::function<std::vector<int>(const dolfinx::graph::AdjacencyList<int>&)>&)’ at /tmp/makepkg/dolfinx/src/dolfinx-0.6.0/cpp/dolfinx/fem/dofmapbuilder.cpp:607:51:
/usr/include/c++/13.1.1/bits/new_allocator.h:168:33: warning: ‘void operator delete(void*, std::size_t)’ called on pointer ‘<unknown>’ with nonzero offset [4, 9223372036854775804] [-Wfree-nonheap-object]
  168 |         _GLIBCXX_OPERATOR_DELETE(_GLIBCXX_SIZED_DEALLOC(__p, __n));
      |                                 ^
In member function ‘_Tp* std::__new_allocator<_Tp>::allocate(size_type, const void*) [with _Tp = int]’,
    inlined from ‘constexpr _Tp* std::allocator< <template-parameter-1-1> >::allocate(std::size_t) [with _Tp = int]’ at /usr/include/c++/13.1.1/bits/allocator.h:198:40,
    inlined from ‘static constexpr _Tp* std::allocator_traits<std::allocator<_Up> >::allocate(allocator_type&, size_type) [with _Tp = int]’ at /usr/include/c++/13.1.1/bits/alloc_traits.h:482:28,
    inlined from ‘constexpr std::_Vector_base<_Tp, _Alloc>::pointer std::_Vector_base<_Tp, _Alloc>::_M_allocate(std::size_t) [with _Tp = int; _Alloc = std::allocator<int>]’ at /usr/include/c++/13.1.1/bits/stl_vector.h:378:33,
    inlined from ‘constexpr std::_Vector_base<_Tp, _Alloc>::pointer std::_Vector_base<_Tp, _Alloc>::_M_allocate(std::size_t) [with _Tp = int; _Alloc = std::allocator<int>]’ at /usr/include/c++/13.1.1/bits/stl_vector.h:375:7,
    inlined from ‘constexpr void std::_Vector_base<_Tp, _Alloc>::_M_create_storage(std::size_t) [with _Tp = int; _Alloc = std::allocator<int>]’ at /usr/include/c++/13.1.1/bits/stl_vector.h:395:44,
    inlined from ‘constexpr std::_Vector_base<_Tp, _Alloc>::_Vector_base(std::size_t, const allocator_type&) [with _Tp = int; _Alloc = std::allocator<int>]’ at /usr/include/c++/13.1.1/bits/stl_vector.h:332:26,
    inlined from ‘constexpr std::vector<_Tp, _Alloc>::vector(size_type, const allocator_type&) [with _Tp = int; _Alloc = std::allocator<int>]’ at /usr/include/c++/13.1.1/bits/stl_vector.h:554:47,
    inlined from ‘std::pair<std::vector<long int>, std::vector<int> > {anonymous}::get_global_indices(const dolfinx::mesh::Topology&, int32_t, int64_t, const std::vector<long int>&, const std::vector<int>&, const std::vector<std::pair<signed char, int> >&)’ at /tmp/makepkg/dolfinx/src/dolfinx-0.6.0/cpp/dolfinx/fem/dofmapbuilder.cpp:480:44,
    inlined from ‘std::tuple<dolfinx::common::IndexMap, int, dolfinx::graph::AdjacencyList<int> > dolfinx::fem::build_dofmap_data(MPI_Comm, const dolfinx::mesh::Topology&, const ElementDofLayout&, const std::function<std::vector<int>(const dolfinx::graph::AdjacencyList<int>&)>&)’ at /tmp/makepkg/dolfinx/src/dolfinx-0.6.0/cpp/dolfinx/fem/dofmapbuilder.cpp:607:51:
/usr/include/c++/13.1.1/bits/new_allocator.h:147:55: note: returned from ‘void* operator new(std::size_t)’
  147 |         return static_cast<_Tp*>(_GLIBCXX_OPERATOR_NEW(__n * sizeof(_Tp)));
      |                                                       ^
[ 38%] Building CXX object dolfinx/CMakeFiles/dolfinx.dir/fem/utils.cpp.o
[ 40%] Building CXX object dolfinx/CMakeFiles/dolfinx.dir/geometry/BoundingBoxTree.cpp.o
[ 42%] Building CXX object dolfinx/CMakeFiles/dolfinx.dir/geometry/gjk.cpp.o
[ 44%] Building CXX object dolfinx/CMakeFiles/dolfinx.dir/geometry/utils.cpp.o
[ 46%] Building CXX object dolfinx/CMakeFiles/dolfinx.dir/graph/ordering.cpp.o
[ 48%] Building CXX object dolfinx/CMakeFiles/dolfinx.dir/graph/partitioners.cpp.o
[ 50%] Building CXX object dolfinx/CMakeFiles/dolfinx.dir/graph/partition.cpp.o
[ 51%] Building CXX object dolfinx/CMakeFiles/dolfinx.dir/io/ADIOS2Writers.cpp.o
[ 53%] Building CXX object dolfinx/CMakeFiles/dolfinx.dir/io/cells.cpp.o
[ 55%] Building CXX object dolfinx/CMakeFiles/dolfinx.dir/io/HDF5Interface.cpp.o
[ 57%] Building CXX object dolfinx/CMakeFiles/dolfinx.dir/io/VTKFile.cpp.o
[ 59%] Building CXX object dolfinx/CMakeFiles/dolfinx.dir/io/vtk_utils.cpp.o
[ 61%] Building CXX object dolfinx/CMakeFiles/dolfinx.dir/io/XDMFFile.cpp.o
[ 63%] Building CXX object dolfinx/CMakeFiles/dolfinx.dir/io/xdmf_function.cpp.o
[ 65%] Building CXX object dolfinx/CMakeFiles/dolfinx.dir/io/xdmf_mesh.cpp.o
[ 67%] Building CXX object dolfinx/CMakeFiles/dolfinx.dir/io/xdmf_utils.cpp.o
[ 69%] Building CXX object dolfinx/CMakeFiles/dolfinx.dir/la/SparsityPattern.cpp.o
[ 71%] Building CXX object dolfinx/CMakeFiles/dolfinx.dir/la/petsc.cpp.o
[ 73%] Building CXX object dolfinx/CMakeFiles/dolfinx.dir/la/slepc.cpp.o
[ 75%] Building CXX object dolfinx/CMakeFiles/dolfinx.dir/mesh/Mesh.cpp.o
[ 76%] Building CXX object dolfinx/CMakeFiles/dolfinx.dir/mesh/Geometry.cpp.o
[ 78%] Building CXX object dolfinx/CMakeFiles/dolfinx.dir/mesh/Topology.cpp.o
[ 80%] Building CXX object dolfinx/CMakeFiles/dolfinx.dir/mesh/cell_types.cpp.o
[ 82%] Building CXX object dolfinx/CMakeFiles/dolfinx.dir/mesh/generation.cpp.o
In file included from /tmp/makepkg/dolfinx/src/dolfinx-0.6.0/cpp/dolfinx/mesh/cell_types.h:11,
                 from /tmp/makepkg/dolfinx/src/dolfinx-0.6.0/cpp/dolfinx/mesh/cell_types.cpp:7:
/tmp/makepkg/dolfinx/src/dolfinx-0.6.0/cpp/dolfinx/graph/AdjacencyList.h:32:37: error: ‘int32_t’ in namespace ‘std’ does not name a type
   32 |   explicit AdjacencyList(const std::int32_t n) : _array(n), _offsets(n + 1)
      |                                     ^~~~~~~
/tmp/makepkg/dolfinx/src/dolfinx-0.6.0/cpp/dolfinx/graph/AdjacencyList.h:43:50: error: ‘int32_t’ is not a member of ‘std’; did you mean ‘int32_t’?
   43 |             std::convertible_to<std::vector<std::int32_t>> V>
      |                                                  ^~~~~~~
In file included from /usr/include/sys/types.h:155,
                 from /usr/include/stdlib.h:395,
                 from /usr/include/c++/13.1.1/cstdlib:79,
                 from /usr/include/c++/13.1.1/ext/string_conversions.h:43,
                 from /usr/include/c++/13.1.1/bits/basic_string.h:4110,
                 from /usr/include/c++/13.1.1/string:54,
                 from /usr/include/c++/13.1.1/bits/locale_classes.h:40,
                 from /usr/include/c++/13.1.1/bits/ios_base.h:41,
                 from /usr/include/c++/13.1.1/ios:44,
                 from /usr/include/c++/13.1.1/istream:40,
                 from /usr/include/c++/13.1.1/sstream:40,
                 from /tmp/makepkg/dolfinx/src/dolfinx-0.6.0/cpp/dolfinx/graph/AdjacencyList.h:13:
/usr/include/bits/stdint-intn.h:26:19: note: ‘int32_t’ declared here
   26 | typedef __int32_t int32_t;
      |                   ^~~~~~~
/tmp/makepkg/dolfinx/src/dolfinx-0.6.0/cpp/dolfinx/graph/AdjacencyList.h:43:50: error: template argument 1 is invalid
   43 |             std::convertible_to<std::vector<std::int32_t>> V>
      |                                                  ^~~~~~~
/tmp/makepkg/dolfinx/src/dolfinx-0.6.0/cpp/dolfinx/graph/AdjacencyList.h:43:50: error: template argument 2 is invalid
/tmp/makepkg/dolfinx/src/dolfinx-0.6.0/cpp/dolfinx/graph/AdjacencyList.h:43:18: error: expected identifier
   43 |             std::convertible_to<std::vector<std::int32_t>> V>
      |                  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/tmp/makepkg/dolfinx/src/dolfinx-0.6.0/cpp/dolfinx/graph/AdjacencyList.h:43:18: error: wrong number of template arguments (1, should be 2)
In file included from /usr/include/c++/13.1.1/compare:37,
                 from /usr/include/c++/13.1.1/array:38,
                 from /tmp/makepkg/dolfinx/src/dolfinx-0.6.0/cpp/dolfinx/mesh/cell_types.h:9:
/usr/include/c++/13.1.1/concepts:72:13: note: provided for ‘template<class _From, class _To> concept std::convertible_to’
   72 |     concept convertible_to = is_convertible_v<_From, _To>
      |             ^~~~~~~~~~~~~~
/tmp/makepkg/dolfinx/src/dolfinx-0.6.0/cpp/dolfinx/graph/AdjacencyList.h:43:60: error: expected unqualified-id before ‘V’
   43 |             std::convertible_to<std::vector<std::int32_t>> V>
      |                                                            ^
/tmp/makepkg/dolfinx/src/dolfinx-0.6.0/cpp/dolfinx/graph/AdjacencyList.h:43:60: error: expected ‘>’ before ‘V’
/tmp/makepkg/dolfinx/src/dolfinx-0.6.0/cpp/dolfinx/graph/AdjacencyList.h:44:27: error: ‘V’ has not been declared
   44 |   AdjacencyList(U&& data, V&& offsets)
      |                           ^
/tmp/makepkg/dolfinx/src/dolfinx-0.6.0/cpp/dolfinx/graph/AdjacencyList.h:94:8: error: ‘int32_t’ in namespace ‘std’ does not name a type
   94 |   std::int32_t num_nodes() const { return _offsets.size() - 1; }
      |        ^~~~~~~
/tmp/makepkg/dolfinx/src/dolfinx-0.6.0/cpp/dolfinx/graph/AdjacencyList.h:132:26: error: ‘int32_t’ is not a member of ‘std’; did you mean ‘int32_t’?
  132 |   const std::vector<std::int32_t>& offsets() const { return _offsets; }
      |                          ^~~~~~~
/usr/include/bits/stdint-intn.h:26:19: note: ‘int32_t’ declared here
   26 | typedef __int32_t int32_t;
      |                   ^~~~~~~
/tmp/makepkg/dolfinx/src/dolfinx-0.6.0/cpp/dolfinx/graph/AdjacencyList.h:132:33: error: template argument 1 is invalid
  132 |   const std::vector<std::int32_t>& offsets() const { return _offsets; }
      |                                 ^
/tmp/makepkg/dolfinx/src/dolfinx-0.6.0/cpp/dolfinx/graph/AdjacencyList.h:132:33: error: template argument 2 is invalid
/tmp/makepkg/dolfinx/src/dolfinx-0.6.0/cpp/dolfinx/graph/AdjacencyList.h:132:14: error: ‘<expression error>’ in namespace ‘std’ does not name a type
  132 |   const std::vector<std::int32_t>& offsets() const { return _offsets; }
      |              ^~~~~~~~~~~~~~~~~~~~
/tmp/makepkg/dolfinx/src/dolfinx-0.6.0/cpp/dolfinx/graph/AdjacencyList.h:135:20: error: ‘int32_t’ is not a member of ‘std’; did you mean ‘int32_t’?
  135 |   std::vector<std::int32_t>& offsets() { return _offsets; }
      |                    ^~~~~~~
/usr/include/bits/stdint-intn.h:26:19: note: ‘int32_t’ declared here
   26 | typedef __int32_t int32_t;
      |                   ^~~~~~~
/tmp/makepkg/dolfinx/src/dolfinx-0.6.0/cpp/dolfinx/graph/AdjacencyList.h:135:27: error: template argument 1 is invalid
  135 |   std::vector<std::int32_t>& offsets() { return _offsets; }
      |                           ^
/tmp/makepkg/dolfinx/src/dolfinx-0.6.0/cpp/dolfinx/graph/AdjacencyList.h:135:27: error: template argument 2 is invalid
/tmp/makepkg/dolfinx/src/dolfinx-0.6.0/cpp/dolfinx/graph/AdjacencyList.h:135:8: error: ‘<expression error>’ in namespace ‘std’ does not name a type
  135 |   std::vector<std::int32_t>& offsets() { return _offsets; }
      |        ^~~~~~~~~~~~~~~~~~~~
/tmp/makepkg/dolfinx/src/dolfinx-0.6.0/cpp/dolfinx/graph/AdjacencyList.h:159:20: error: ‘int32_t’ is not a member of ‘std’; did you mean ‘int32_t’?
  159 |   std::vector<std::int32_t> _offsets;
      |                    ^~~~~~~
/usr/include/bits/stdint-intn.h:26:19: note: ‘int32_t’ declared here
   26 | typedef __int32_t int32_t;
      |                   ^~~~~~~
/tmp/makepkg/dolfinx/src/dolfinx-0.6.0/cpp/dolfinx/graph/AdjacencyList.h:159:27: error: template argument 1 is invalid
  159 |   std::vector<std::int32_t> _offsets;
      |                           ^
/tmp/makepkg/dolfinx/src/dolfinx-0.6.0/cpp/dolfinx/graph/AdjacencyList.h:159:27: error: template argument 2 is invalid
/tmp/makepkg/dolfinx/src/dolfinx-0.6.0/cpp/dolfinx/graph/AdjacencyList.h:159:8: error: ‘<expression error>’ in namespace ‘std’ does not name a type
  159 |   std::vector<std::int32_t> _offsets;
      |        ^~~~~~~~~~~~~~~~~~~~
/tmp/makepkg/dolfinx/src/dolfinx-0.6.0/cpp/dolfinx/graph/AdjacencyList.h: In constructor ‘dolfinx::graph::AdjacencyList<T>::AdjacencyList(int)’:
/tmp/makepkg/dolfinx/src/dolfinx-0.6.0/cpp/dolfinx/graph/AdjacencyList.h:32:61: error: class ‘dolfinx::graph::AdjacencyList<T>’ does not have any field named ‘_offsets’
   32 |   explicit AdjacencyList(const std::int32_t n) : _array(n), _offsets(n + 1)
      |                                                             ^~~~~~~~
/tmp/makepkg/dolfinx/src/dolfinx-0.6.0/cpp/dolfinx/graph/AdjacencyList.h:35:15: error: ‘_offsets’ was not declared in this scope
   35 |     std::iota(_offsets.begin(), _offsets.end(), 0);
      |               ^~~~~~~~
/tmp/makepkg/dolfinx/src/dolfinx-0.6.0/cpp/dolfinx/graph/AdjacencyList.h: In constructor ‘dolfinx::graph::AdjacencyList<T>::AdjacencyList(U&&, int&&)’:
/tmp/makepkg/dolfinx/src/dolfinx-0.6.0/cpp/dolfinx/graph/AdjacencyList.h:45:40: error: class ‘dolfinx::graph::AdjacencyList<T>’ does not have any field named ‘_offsets’
   45 |       : _array(std::forward<U>(data)), _offsets(std::forward<V>(offsets))
      |                                        ^~~~~~~~
/tmp/makepkg/dolfinx/src/dolfinx-0.6.0/cpp/dolfinx/graph/AdjacencyList.h:45:62: error: ‘V’ was not declared in this scope
   45 |       : _array(std::forward<U>(data)), _offsets(std::forward<V>(offsets))
      |                                                              ^
/tmp/makepkg/dolfinx/src/dolfinx-0.6.0/cpp/dolfinx/graph/AdjacencyList.h:47:20: error: ‘_offsets’ was not declared in this scope; did you mean ‘offsets’?
   47 |     _array.reserve(_offsets.back());
      |                    ^~~~~~~~
      |                    offsets
In file included from /usr/include/c++/13.1.1/cassert:44,
                 from /tmp/makepkg/dolfinx/src/dolfinx-0.6.0/cpp/dolfinx/graph/AdjacencyList.h:9:
/tmp/makepkg/dolfinx/src/dolfinx-0.6.0/cpp/dolfinx/graph/AdjacencyList.h:48:37: error: ‘int32_t’ is not a member of ‘std’; did you mean ‘int32_t’?
   48 |     assert(_offsets.back() == (std::int32_t)_array.size());
      |                                     ^~~~~~~
/usr/include/bits/stdint-intn.h:26:19: note: ‘int32_t’ declared here
   26 | typedef __int32_t int32_t;
      |                   ^~~~~~~
/tmp/makepkg/dolfinx/src/dolfinx-0.6.0/cpp/dolfinx/graph/AdjacencyList.h:48:45: error: expected ‘)’ before ‘_array’
   48 |     assert(_offsets.back() == (std::int32_t)_array.size());
      |                                             ^~~~~~
/tmp/makepkg/dolfinx/src/dolfinx-0.6.0/cpp/dolfinx/graph/AdjacencyList.h:48:45: error: expected ‘)’ before ‘_array’
/tmp/makepkg/dolfinx/src/dolfinx-0.6.0/cpp/dolfinx/graph/AdjacencyList.h:48:5: note: to match this ‘(’
   48 |     assert(_offsets.back() == (std::int32_t)_array.size());
      |     ^~~~~~
/tmp/makepkg/dolfinx/src/dolfinx-0.6.0/cpp/dolfinx/graph/AdjacencyList.h: In constructor ‘dolfinx::graph::AdjacencyList<T>::AdjacencyList(const std::vector<X>&)’:
/tmp/makepkg/dolfinx/src/dolfinx-0.6.0/cpp/dolfinx/graph/AdjacencyList.h:60:5: error: ‘_offsets’ was not declared in this scope
   60 |     _offsets.reserve(data.size() + 1);
      |     ^~~~~~~~
/tmp/makepkg/dolfinx/src/dolfinx-0.6.0/cpp/dolfinx/graph/AdjacencyList.h: In member function ‘int dolfinx::graph::AdjacencyList<T>::num_links(int) const’:
/tmp/makepkg/dolfinx/src/dolfinx-0.6.0/cpp/dolfinx/graph/AdjacencyList.h:101:30: error: ‘_offsets’ was not declared in this scope
  101 |     assert((node + 1) < (int)_offsets.size());
      |                              ^~~~~~~~
/tmp/makepkg/dolfinx/src/dolfinx-0.6.0/cpp/dolfinx/graph/AdjacencyList.h: In member function ‘std::span<_Type, 1844674[407](https://gitlab.com/dune-archiso/testing/aur/aur/-/jobs/4240481630#L407)37095[516](https://gitlab.com/dune-archiso/testing/aur/aur/-/jobs/4240481630#L516)15> dolfinx::graph::AdjacencyList<T>::links(int)’:
/tmp/makepkg/dolfinx/src/dolfinx-0.6.0/cpp/dolfinx/graph/AdjacencyList.h:111:41: error: ‘_offsets’ was not declared in this scope
  111 |     return std::span<T>(_array.data() + _offsets[node],
      |                                         ^~~~~~~~
/tmp/makepkg/dolfinx/src/dolfinx-0.6.0/cpp/dolfinx/graph/AdjacencyList.h: In member function ‘std::span<const T> dolfinx::graph::AdjacencyList<T>::links(int) const’:
/tmp/makepkg/dolfinx/src/dolfinx-0.6.0/cpp/dolfinx/graph/AdjacencyList.h:121:47: error: ‘_offsets’ was not declared in this scope
  121 |     return std::span<const T>(_array.data() + _offsets[node],
      |                                               ^~~~~~~~
/tmp/makepkg/dolfinx/src/dolfinx-0.6.0/cpp/dolfinx/graph/AdjacencyList.h: In member function ‘std::string dolfinx::graph::AdjacencyList<T>::str() const’:
/tmp/makepkg/dolfinx/src/dolfinx-0.6.0/cpp/dolfinx/graph/AdjacencyList.h:144:33: error: ‘_offsets’ was not declared in this scope
  144 |     for (std::size_t e = 0; e < _offsets.size() - 1; ++e)
      |                                 ^~~~~~~~
/tmp/makepkg/dolfinx/src/dolfinx-0.6.0/cpp/dolfinx/graph/AdjacencyList.h: In function ‘dolfinx::graph::AdjacencyList<typename std::decay<_Tp>::type::value_type> dolfinx::graph::regular_adjacency_list(U&&, int)’:
/tmp/makepkg/dolfinx/src/dolfinx-0.6.0/cpp/dolfinx/graph/AdjacencyList.h:190:8: error: ‘int32_t’ is not a member of ‘std’; did you mean ‘int32_t’?
  190 |   std::int32_t num_nodes = degree == 0 ? data.size() : data.size() / degree;
      |        ^~~~~~~
/usr/include/bits/stdint-intn.h:26:19: note: ‘int32_t’ declared here
   26 | typedef __int32_t int32_t;
      |                   ^~~~~~~
/tmp/makepkg/dolfinx/src/dolfinx-0.6.0/cpp/dolfinx/graph/AdjacencyList.h:191:20: error: ‘int32_t’ is not a member of ‘std’; did you mean ‘int32_t’?
  191 |   std::vector<std::int32_t> offsets(num_nodes + 1, 0);
      |                    ^~~~~~~
/usr/include/bits/stdint-intn.h:26:19: note: ‘int32_t’ declared here
   26 | typedef __int32_t int32_t;
      |                   ^~~~~~~
/tmp/makepkg/dolfinx/src/dolfinx-0.6.0/cpp/dolfinx/graph/AdjacencyList.h:191:27: error: template argument 1 is invalid
  191 |   std::vector<std::int32_t> offsets(num_nodes + 1, 0);
      |                           ^
/tmp/makepkg/dolfinx/src/dolfinx-0.6.0/cpp/dolfinx/graph/AdjacencyList.h:191:27: error: template argument 2 is invalid
/tmp/makepkg/dolfinx/src/dolfinx-0.6.0/cpp/dolfinx/graph/AdjacencyList.h:191:37: error: ‘num_nodes’ was not declared in this scope
  191 |   std::vector<std::int32_t> offsets(num_nodes + 1, 0);
      |                                     ^~~~~~~~~
/tmp/makepkg/dolfinx/src/dolfinx-0.6.0/cpp/dolfinx/graph/AdjacencyList.h:191:53: error: expression list treated as compound expression in initializer [-fpermissive]
  191 |   std::vector<std::int32_t> offsets(num_nodes + 1, 0);
      |                                                     ^
/tmp/makepkg/dolfinx/src/dolfinx-0.6.0/cpp/dolfinx/graph/AdjacencyList.h:192:39: error: request for member ‘size’ in ‘offsets’, which is of non-class type ‘int’
  192 |   for (std::size_t i = 1; i < offsets.size(); ++i)
      |                                       ^~~~
/tmp/makepkg/dolfinx/src/dolfinx-0.6.0/cpp/dolfinx/graph/AdjacencyList.h:193:12: error: invalid types ‘int[std::size_t {aka long unsigned int}]’ for array subscript
  193 |     offsets[i] = offsets[i - 1] + degree;
      |            ^
/tmp/makepkg/dolfinx/src/dolfinx-0.6.0/cpp/dolfinx/graph/AdjacencyList.h:193:25: error: invalid types ‘int[std::size_t {aka long unsigned int}]’ for array subscript
  193 |     offsets[i] = offsets[i - 1] + degree;
      |                         ^
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/Makefile2:304: dolfinx/CMakeFiles/dolfinx.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
```

**Comment by chrisrichardson** [comment]
I think this is due to missing 
```
#include <cstdint>
```

from a few files

**Comment by chrisrichardson** [comment]
Looks like `AdjacencyList.h` and `xdmf_mesh.h` need it

**Comment by carlosal1015** [comment]
Thanks 👍🏼 for the hint, it is [fixed now](https://aur.archlinux.org/cgit/aur.git/commit/?h=dolfinx&id=ff98c08db148538e2076dc91b1bab64c1bcab90c) as the same with [basix](https://aur.archlinux.org/cgit/aur.git/commit/?h=basix&id=6fad6eab9fdaec4b5b43a227907f0e323c76fd4d).

```
-- Configuring done (6.7s)
-- Generating done (0.0s)
-- Build files have been written to: /tmp/makepkg/dolfinx/src/build
[  1%] Building CXX object dolfinx/CMakeFiles/dolfinx.dir/common/defines.cpp.o
[  3%] Building CXX object dolfinx/CMakeFiles/dolfinx.dir/common/IndexMap.cpp.o
[  5%] Building CXX object dolfinx/CMakeFiles/dolfinx.dir/common/MPI.cpp.o
[  7%] Building CXX object dolfinx/CMakeFiles/dolfinx.dir/common/log.cpp.o
[  9%] Building CXX object dolfinx/CMakeFiles/dolfinx.dir/common/Table.cpp.o
[ 11%] Building CXX object dolfinx/CMakeFiles/dolfinx.dir/common/Timer.cpp.o
[ 13%] Building CXX object dolfinx/CMakeFiles/dolfinx.dir/common/TimeLogger.cpp.o
[ 15%] Building CXX object dolfinx/CMakeFiles/dolfinx.dir/common/TimeLogManager.cpp.o
[ 17%] Building CXX object dolfinx/CMakeFiles/dolfinx.dir/common/timing.cpp.o
[ 19%] Building CXX object dolfinx/CMakeFiles/dolfinx.dir/fem/DirichletBC.cpp.o
[ 21%] Building CXX object dolfinx/CMakeFiles/dolfinx.dir/fem/CoordinateElement.cpp.o
[ 23%] Building CXX object dolfinx/CMakeFiles/dolfinx.dir/fem/DofMap.cpp.o
[ 25%] Building CXX object dolfinx/CMakeFiles/dolfinx.dir/fem/ElementDofLayout.cpp.o
[ 26%] Building CXX object dolfinx/CMakeFiles/dolfinx.dir/fem/FiniteElement.cpp.o
[ 28%] Building CXX object dolfinx/CMakeFiles/dolfinx.dir/fem/FunctionSpace.cpp.o
[ 30%] Building CXX object dolfinx/CMakeFiles/dolfinx.dir/fem/dofmapbuilder.cpp.o
[ 32%] Building CXX object dolfinx/CMakeFiles/dolfinx.dir/fem/interpolate.cpp.o
[ 34%] Building CXX object dolfinx/CMakeFiles/dolfinx.dir/fem/petsc.cpp.o
[ 36%] Building CXX object dolfinx/CMakeFiles/dolfinx.dir/fem/sparsitybuild.cpp.o
[ 38%] Building CXX object dolfinx/CMakeFiles/dolfinx.dir/fem/utils.cpp.o
In file included from /usr/include/c++/13.1.1/x86_64-pc-linux-gnu/bits/c++allocator.h:33,
                 from /usr/include/c++/13.1.1/bits/allocator.h:46,
                 from /usr/include/c++/13.1.1/bits/alloc_traits.h:39,
                 from /usr/include/c++/13.1.1/ext/alloc_traits.h:34,
                 from /usr/include/c++/13.1.1/bits/hashtable_policy.h:39,
                 from /usr/include/c++/13.1.1/bits/hashtable.h:35,
                 from /usr/include/c++/13.1.1/bits/unordered_map.h:33,
                 from /usr/include/c++/13.1.1/unordered_map:41,
                 from /usr/include/c++/13.1.1/functional:63,
                 from /tmp/makepkg/dolfinx/src/dolfinx-0.6.0/cpp/dolfinx/fem/dofmapbuilder.h:10,
                 from /tmp/makepkg/dolfinx/src/dolfinx-0.6.0/cpp/dolfinx/fem/dofmapbuilder.cpp:7:
In member function ‘void std::__new_allocator<_Tp>::deallocate(_Tp*, size_type) [with _Tp = int]’,
    inlined from ‘constexpr void std::allocator< <template-parameter-1-1> >::deallocate(_Tp*, std::size_t) [with _Tp = int]’ at /usr/include/c++/13.1.1/bits/allocator.h:210:35,
    inlined from ‘static constexpr void std::allocator_traits<std::allocator<_Up> >::deallocate(allocator_type&, pointer, size_type) [with _Tp = int]’ at /usr/include/c++/13.1.1/bits/alloc_traits.h:516:23,
    inlined from ‘constexpr void std::_Vector_base<_Tp, _Alloc>::_M_deallocate(pointer, std::size_t) [with _Tp = int; _Alloc = std::allocator<int>]’ at /usr/include/c++/13.1.1/bits/stl_vector.h:387:19,
    inlined from ‘constexpr void std::vector<_Tp, _Alloc>::reserve(size_type) [with _Tp = int; _Alloc = std::allocator<int>]’ at /usr/include/c++/13.1.1/bits/vector.tcc:93:17,
    inlined from ‘std::pair<std::vector<long int>, std::vector<int> > {anonymous}::get_global_indices(const dolfinx::mesh::Topology&, int32_t, int64_t, const std::vector<long int>&, const std::vector<int>&, const std::vector<std::pair<signed char, int> >&)’ at /tmp/makepkg/dolfinx/src/dolfinx-0.6.0/cpp/dolfinx/fem/dofmapbuilder.cpp:481:24,
    inlined from ‘std::tuple<dolfinx::common::IndexMap, int, dolfinx::graph::AdjacencyList<int> > dolfinx::fem::build_dofmap_data(MPI_Comm, const dolfinx::mesh::Topology&, const ElementDofLayout&, const std::function<std::vector<int>(const dolfinx::graph::AdjacencyList<int>&)>&)’ at /tmp/makepkg/dolfinx/src/dolfinx-0.6.0/cpp/dolfinx/fem/dofmapbuilder.cpp:607:51:
/usr/include/c++/13.1.1/bits/new_allocator.h:168:33: warning: ‘void operator delete(void*, std::size_t)’ called on pointer ‘<unknown>’ with nonzero offset [4, 922337203685[477](https://gitlab.com/dune-archiso/testing/aur/aur/-/jobs/4240499311#L477)5804] [-Wfree-nonheap-object]
  168 |         _GLIBCXX_OPERATOR_DELETE(_GLIBCXX_SIZED_DEALLOC(__p, __n));
      |                                 ^
In member function ‘_Tp* std::__new_allocator<_Tp>::allocate(size_type, const void*) [with _Tp = int]’,
    inlined from ‘constexpr _Tp* std::allocator< <template-parameter-1-1> >::allocate(std::size_t) [with _Tp = int]’ at /usr/include/c++/13.1.1/bits/allocator.h:198:40,
    inlined from ‘static constexpr _Tp* std::allocator_traits<std::allocator<_Up> >::allocate(allocator_type&, size_type) [with _Tp = int]’ at /usr/include/c++/13.1.1/bits/alloc_traits.h:482:28,
    inlined from ‘constexpr std::_Vector_base<_Tp, _Alloc>::pointer std::_Vector_base<_Tp, _Alloc>::_M_allocate(std::size_t) [with _Tp = int; _Alloc = std::allocator<int>]’ at /usr/include/c++/13.1.1/bits/stl_vector.h:378:33,
    inlined from ‘constexpr std::_Vector_base<_Tp, _Alloc>::pointer std::_Vector_base<_Tp, _Alloc>::_M_allocate(std::size_t) [with _Tp = int; _Alloc = std::allocator<int>]’ at /usr/include/c++/13.1.1/bits/stl_vector.h:375:7,
    inlined from ‘constexpr void std::_Vector_base<_Tp, _Alloc>::_M_create_storage(std::size_t) [with _Tp = int; _Alloc = std::allocator<int>]’ at /usr/include/c++/13.1.1/bits/stl_vector.h:395:44,
    inlined from ‘constexpr std::_Vector_base<_Tp, _Alloc>::_Vector_base(std::size_t, const allocator_type&) [with _Tp = int; _Alloc = std::allocator<int>]’ at /usr/include/c++/13.1.1/bits/stl_vector.h:332:26,
    inlined from ‘constexpr std::vector<_Tp, _Alloc>::vector(size_type, const allocator_type&) [with _Tp = int; _Alloc = std::allocator<int>]’ at /usr/include/c++/13.1.1/bits/stl_vector.h:554:47,
    inlined from ‘std::pair<std::vector<long int>, std::vector<int> > {anonymous}::get_global_indices(const dolfinx::mesh::Topology&, int32_t, int64_t, const std::vector<long int>&, const std::vector<int>&, const std::vector<std::pair<signed char, int> >&)’ at /tmp/makepkg/dolfinx/src/dolfinx-0.6.0/cpp/dolfinx/fem/dofmapbuilder.cpp:[480](https://gitlab.com/dune-archiso/testing/aur/aur/-/jobs/4240499311#L480):44,
    inlined from ‘std::tuple<dolfinx::common::IndexMap, int, dolfinx::graph::AdjacencyList<int> > dolfinx::fem::build_dofmap_data(MPI_Comm, const dolfinx::mesh::Topology&, const ElementDofLayout&, const std::function<std::vector<int>(const dolfinx::graph::AdjacencyList<int>&)>&)’ at /tmp/makepkg/dolfinx/src/dolfinx-0.6.0/cpp/dolfinx/fem/dofmapbuilder.cpp:607:51:
/usr/include/c++/13.1.1/bits/new_allocator.h:147:55: note: returned from ‘void* operator new(std::size_t)’
  147 |         return static_cast<_Tp*>(_GLIBCXX_OPERATOR_NEW(__n * sizeof(_Tp)));
      |                                                       ^
[ 40%] Building CXX object dolfinx/CMakeFiles/dolfinx.dir/geometry/BoundingBoxTree.cpp.o
[ 42%] Building CXX object dolfinx/CMakeFiles/dolfinx.dir/geometry/gjk.cpp.o
[ 44%] Building CXX object dolfinx/CMakeFiles/dolfinx.dir/geometry/utils.cpp.o
[ 46%] Building CXX object dolfinx/CMakeFiles/dolfinx.dir/graph/ordering.cpp.o
[ 48%] Building CXX object dolfinx/CMakeFiles/dolfinx.dir/graph/partitioners.cpp.o
[ 50%] Building CXX object dolfinx/CMakeFiles/dolfinx.dir/graph/partition.cpp.o
[ 51%] Building CXX object dolfinx/CMakeFiles/dolfinx.dir/io/ADIOS2Writers.cpp.o
[ 53%] Building CXX object dolfinx/CMakeFiles/dolfinx.dir/io/cells.cpp.o
[ 55%] Building CXX object dolfinx/CMakeFiles/dolfinx.dir/io/HDF5Interface.cpp.o
[ 57%] Building CXX object dolfinx/CMakeFiles/dolfinx.dir/io/VTKFile.cpp.o
[ 59%] Building CXX object dolfinx/CMakeFiles/dolfinx.dir/io/vtk_utils.cpp.o
[ 61%] Building CXX object dolfinx/CMakeFiles/dolfinx.dir/io/XDMFFile.cpp.o
[ 63%] Building CXX object dolfinx/CMakeFiles/dolfinx.dir/io/xdmf_function.cpp.o
[ 65%] Building CXX object dolfinx/CMakeFiles/dolfinx.dir/io/xdmf_mesh.cpp.o
[ 67%] Building CXX object dolfinx/CMakeFiles/dolfinx.dir/io/xdmf_utils.cpp.o
[ 69%] Building CXX object dolfinx/CMakeFiles/dolfinx.dir/la/SparsityPattern.cpp.o
[ 71%] Building CXX object dolfinx/CMakeFiles/dolfinx.dir/la/petsc.cpp.o
[ 73%] Building CXX object dolfinx/CMakeFiles/dolfinx.dir/la/slepc.cpp.o
[ 75%] Building CXX object dolfinx/CMakeFiles/dolfinx.dir/mesh/Mesh.cpp.o
[ 76%] Building CXX object dolfinx/CMakeFiles/dolfinx.dir/mesh/Geometry.cpp.o
[ 78%] Building CXX object dolfinx/CMakeFiles/dolfinx.dir/mesh/Topology.cpp.o
[ 80%] Building CXX object dolfinx/CMakeFiles/dolfinx.dir/mesh/cell_types.cpp.o
[ 82%] Building CXX object dolfinx/CMakeFiles/dolfinx.dir/mesh/generation.cpp.o
[ 84%] Building CXX object dolfinx/CMakeFiles/dolfinx.dir/mesh/graphbuild.cpp.o
[ 86%] Building CXX object dolfinx/CMakeFiles/dolfinx.dir/mesh/permutationcomputation.cpp.o
[ 88%] Building CXX object dolfinx/CMakeFiles/dolfinx.dir/mesh/topologycomputation.cpp.o
[ 90%] Building CXX object dolfinx/CMakeFiles/dolfinx.dir/mesh/utils.cpp.o
[ 92%] Building CXX object dolfinx/CMakeFiles/dolfinx.dir/nls/NewtonSolver.cpp.o
[ 94%] Building CXX object dolfinx/CMakeFiles/dolfinx.dir/refinement/plaza.cpp.o
[ 96%] Building CXX object dolfinx/CMakeFiles/dolfinx.dir/refinement/refine.cpp.o
[ 98%] Building CXX object dolfinx/CMakeFiles/dolfinx.dir/refinement/utils.cpp.o
[100%] Linking CXX shared library libdolfinx.so
[100%] Built target dolfinx
```

Feel free to close the issue.

**Comment by chrisrichardson** [comment]
Thanks @carlosal1015
